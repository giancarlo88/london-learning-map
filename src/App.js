import React, { Component } from 'react'
import './App.css'
import MapContainer from './Containers/MapContainer/MapContainer'
import AddMarkerContainer from './Containers/AddMarkerContainer/AddMarkerContainer'

class App extends Component {
  constructor() {
    super()
    this.state = {
      isAddingMarker: false
    }
  }

  toggleAdminPanel = () =>
    this.setState({ isAddingMarker: !this.state.isAddingMarker })

  render() {
    const { isAddingMarker } = this.state
    return isAddingMarker ? (
      <AddMarkerContainer toggleAdminPanel={this.toggleAdminPanel} />
    ) : (
      <MapContainer toggleAdminPanel={this.toggleAdminPanel} />
    )
  }
}

export default App
