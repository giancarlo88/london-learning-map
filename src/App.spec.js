import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

describe(App.name, () => {
  describe('@renders', () => {
    describe('with default conditions', () => {
      const wrapper = shallow(<App />)
      it('should render', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('should render a MapContainer', () => {
        expect(wrapper.find('MapContainer')).toHaveLength(1)
      })
    })

    describe('when state property isAddingMarker is true', () => {
      it('should render an AddMarkerContainer', () => {
        const wrapper = shallow(<App />)
        wrapper.setState({ isAddingMarker: true })
        expect(wrapper.find('AddMarkerContainer')).toHaveLength(1)
      })
    })
  })

  describe('@methods', () => {
    describe('toggleAdminPanel', () => {
      it('should change the isAddingMarker state property', () => {
        const wrapper = shallow(<App />)
        expect(wrapper.state().isAddingMarker).toBeFalsy()
        wrapper.instance().toggleAdminPanel()
        expect(wrapper.state().isAddingMarker).toBeTruthy()
      })
    })
  })
})
