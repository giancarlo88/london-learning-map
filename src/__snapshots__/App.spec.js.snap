// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App @renders with default conditions should render 1`] = `
<MapContainer
  toggleAdminPanel={[Function]}
/>
`;
